### PR TITLE
REPO-1126 - Task Master Integration

### PR DESCRIPTION
**Pull request by prdanelli**

### CODE REVIEW

File: config/features.rb
Comment (fradeve) [inline config/features.rb:22]: I am not sure what this `feature` is for, but this is going to become an integral part of our repository setup and will be enabled for every repository we run -- ever. So won't ever need to disable it.
As far as I am concerned, this can be removed :)

Comment (prdanelli) [reply config/features.rb:22]: @fradeve I was thinking more of other people using the HA codebase that didn't have this setup in mind. as this feature is very specific to Ubiquity.

Comment (cjcolvar) [reply config/features.rb:22]: :+1: This should get pulled out to the UP specific engine whenever we make that or it could get pulled out into its own specific little gem.

Comment (prdanelli) [reply config/features.rb:22]: I was thinking the same thing. It would be best in its own Gem probably, but i'll carry on developing inside HA for now.

File: app/actors/hyku_addons/actors/member_collection_from_admin_set_actor.rb
Comment (cjcolvar) [inline app/actors/hyku_addons/actors/member_collection_from_admin_set_actor.rb:25]: :+1:

File: Gemfile.lock
Comment (cjcolvar) [inline Gemfile.lock:511]: Was there a change to the Gemfile that brought these in?  Just checking that everything got committed that should be.

File: app/services/hyku_addons/task_master/publish_service.rb
Comment (cjcolvar) [inline app/services/hyku_addons/task_master/publish_service.rb:61]: I wonder if we should pipe this through `Settings` as well.  This isn't vital right now just think it might be good to do eventually.

File: config/features.rb
Comment (cjcolvar) [inline config/features.rb:24]: What do you think about setting this based upon the ENV?  This way we don't have to remember to turn it on for every tenant.
```suggestion
          default: ENV["PUBSUB_SERVICEACCOUNT_KEY"].present?,
```

File: spec/factories/works.rb
Comment (cjcolvar) [inline spec/factories/works.rb:55]: Do you think we could come up with a more generic name for this factory?

File: spec/actors/hyku_addons/actors/task_master_work_actor_spec.rb
Comment (cjcolvar) [inline spec/actors/hyku_addons/actors/task_master_work_actor_spec.rb:1]: Looks like this spec could be removed in favor of `spec/actors/hyku_addons/actors/task_master/work_actor_spec.rb` above.

**Review by cjcolvar — APPROVED** [review]
This looks great!  I have a few questions/suggestions but nothing that I think is really blocking this from being merged.

**Comment by fradeve** [comment]
@prdanelli please let's remember to squash the commits before merging... the history on this branch is getting bigger :)

**Comment by fradeve** [comment]
@prdanelli very very minor, but the name of the service is `taskmaster` rather than `Task Master` or `task_master`

**Comment by prdanelli** [comment]
> @prdanelli very very minor, but the name of the service is `taskmaster` rather than `Task Master` or `task_master`

@fradeve so it would be `Taskmaster` if we are naming classes?

**Review by cjcolvar — APPROVED** [review]

**Pull request merged**
